mucommander-core/src/main/java/com/mucommander/Application.java

```diff
@@ -451,52 +499,75 @@ private void run() {
-            } else {
-                WindowManager.createNewMainFrame(new DefaultMainFramesBuilder());
-            }
-            // Done launching, wake up threads waiting for the application being launched.
-            // Important: this must be done before disposing the splash screen, as this would otherwise create a
-            // deadlock if the AWT event thread were waiting in #waitUntilLaunched .
-            synchronized (LAUNCH_LOCK) {
-                isLaunching = false;
-                LAUNCH_LOCK.notifyAll();
-            }
+            // Invoke in a different thread: https://www.oracle.com/technical-resources/articles/javase/swingworker.html
+            Thread mainThread = new Thread(() -> {
+                LOGGER.error("muC UI about to be presented");
+                // Creates the initial main frame using any initial path specified by the command line.
+                printStartupMessage("Initializing window...");
+                LOGGER.error("folders init");
+                List<String> folders = activator.getInitialFolders();
+                LOGGER.error("muC new main frame to start");
+                if (CollectionUtils.isNotEmpty(folders)) {
+                    WindowManager.createNewMainFrame(new CommandLineMainFrameBuilder(folders));
+                } else {
+                    WindowManager.createNewMainFrame(new DefaultMainFramesBuilder());
+                }
+                LOGGER.error("muC UI presented after: {} ms", ManagementFactory.getRuntimeMXBean().getUptime());
+
+                // Done launching, wake up threads waiting for the application being launched.
+                // Important: this must be done before disposing the splash screen, as this would otherwise create a
+                // deadlock if the AWT event thread were waiting in #waitUntilLaunched .
+                synchronized (LAUNCH_LOCK) {
+                    isLaunching = false;
+                    LAUNCH_LOCK.notifyAll();
+                }
+                LOGGER.error("Launch lock freed");
 
-            // Enable system notifications, only after MainFrame is created as SystemTrayNotifier needs to retrieve
-            // a MainFrame instance
-            if (MuConfigurations.getPreferences()
-                    .getVariable(MuPreference.ENABLE_SYSTEM_NOTIFICATIONS,
-                            MuPreferences.DEFAULT_ENABLE_SYSTEM_NOTIFICATIONS)) {
-                printStartupMessage("Enabling system notifications...");
-                if (com.mucommander.ui.notifier.NotifierProvider.isAvailable())
-                    com.mucommander.ui.notifier.NotifierProvider.getNotifier().setEnabled(true);
-            }
+                // Dispose splash screen.
+                var localSplashScreen = splashScreen;
+                if (localSplashScreen != null) {
+                    localSplashScreen.dispose();
+                    splashScreen = null;
```

Comment:
can we make `splashScreen` a local variable instead?

Reply:
I guess not, it is being used by Application#printStartupMessage

Reply:
maybe we can inject it to printStartupMessage as it's called from a single method, need to try

Reply:
will you try? I mean, who's gonna try :) 
and I'm not sure how to inject that splash instance into printStartupMessage

Reply:
sure, posted https://github.com/mucommander/mucommander/pull/1071

Reply:
ok, I get it - I was thinking that you had some really advanced magic on your mind.

Reply:
PR updated